src/org/chartjs/HorizontalBarDAOChartView.js

```diff
@@ -0,0 +1,138 @@
+                      "January", "February", "March", "April", "May", "June",
+                      "July", "August", "September", "October", "November", "December"
+                    ];
+                    const quarterNames = ['Q1', 'Q2', 'Q3', 'Q4'];
+
+                    switch (self.dateFrequency) {
+                      case net.nanopay.liquidity.ui.dashboard.DateFrequency.MONTHLY:
+                        return `${monthNames[Number.parseInt(dateArray[0])]} ${dateArray[2]}`
+
+                      case net.nanopay.liquidity.ui.dashboard.DateFrequency.QUARTERLY:
+                        return `${quarterNames[Number.parseInt(dateArray[0]) / 3 - 1]} ${dateArray[2]}`
+
+                      case net.nanopay.liquidity.ui.dashboard.DateFrequency.ANNUALLY:
+                        return dateArray[2];
+
+                      default:
+                        return value;
+                    }
+                  }
+                }
+              }],
+              xAxes: [{
+                ticks: {
+                  callback: function (value, index, values) {
+                    return `$${value}`;
+                  }
+                }
+              }]
+            }
+          },
+        };
+      }
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'keyExpr',
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'xExpr',
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'yExpr',
+    }
+  ],
+
+  reactions: [
+    ['', 'propertyChange.yExpr', 'dataUpdate'],
+    ['', 'propertyChange.data', 'dataUpdate']
+  ],
+
+  listeners: [
+    {
+      name: 'dataUpdate',
+      isFramed: true,
+      code: function () {
+        var self = this;
+        self.data
+          .orderBy(this.yExpr)
+          .select(this.GROUP_BY(this.yExpr, this.GROUP_BY(this.keyExpr, this.SUM(this.xExpr))))
+          .then(function (sink) {
+            // Clear data before cloning because it gets clobbered anyway.
+            self.config.data = { datasets: [] };
+            var config = foam.Object.clone(self.config);
+
+            var dataMap = sink.groupKeys.reduce((map, y) => {
+              return sink.groups[y].groupKeys.reduce((map, k) => {
+                map[k] = [];
+                return map;
+              }, map);
+            }, {});
+
+            sink.groupKeys.forEach((y, yi) => {
+              Object.keys(dataMap).forEach(k => {
+                dataMap[k][yi] = 0;
+              });
+              sink.groups[y].groupKeys.forEach(k => {
+                dataMap[k][yi] = sink.groups[y].groups[k].value;
+              });
+            });
+
+            config.data = {
+              labels: sink.groupKeys,
+              datasets: Object.keys(dataMap).map(k => {
+                var data = {
+                  label: k,
+                  data: dataMap[k]
+                };
+                // var style = self.customDatasetStyling[k] || {};
```

Comment:
You might want to preserve this to get different styled bars. Was it causing problems?

Reply:
Done.